### PR TITLE
Fix tree shaking

### PR DESCRIPTION
**Pull request by 0x009922**
## Changes

- Remove a lot of unnecessary SFC names declaration
- `ui` library now has unit & after-build tests driven by `vitest`
- After-build tests ensure tree-shakeability, e.g. library does not have any side-effects.

For other details please see changesets.

## Explorations

- TypeScript `enum`s are evil. They are not tree-shakeable because they are compiled to IIFE:

  ```ts
  enum Status {
    Info
  }
  ```

  ```js
  var Status;
  (function (Status) {
    Status[Status["Info"] = 0] = "Info";
  })(Status || (Status = {}));
  ```

  The way I used to work-around it:

  ```ts
  export const Status = {
    Info: 'info',
    Success: 'success',
    Warning: 'warning',
    Error: 'error',
  } as const

  export type Status = typeof Status[keyof typeof Status]

  function acceptAnyStatus(status: Status) {}

  acceptAnyStatus(Status.Error)

  function acceptInfoOnly(info: typeof Status.Info) {}

  acceptInfoOnly(Status.Info)
   ```

- `jsoneditor` and `lodash` (**edit:**not sure) are evil. They have side-effects. Fortunately, they are only used within `SJsonInput`. I excluded this component from the bundle temporarily - anyway it is unused and out of design system.
- `body-scroll-lock` is evil too, but only a bit. It has a small side-effect check code. I think it can be implemented in pure manner, and maybe I'll open a PR. **upd:** https://github.com/willmcpo/body-scroll-lock/issues/254 and https://github.com/willmcpo/body-scroll-lock/pull/255
  For now, I exclude it from `SModal` component, but I add a new component - `SBodyScrollLockProvider`. `SModal` will use it's locker by default. For details please see sources and `SModal.spec.cy` for usage example (it uses `body-scroll-lock` small wrap).
- `defineComponent()` should always be used with `/* @__PURE__ */` or `/* #__PURE__ */` prefix. Generally, it should be avoided.
- `vite.config.ts` → `build.target` set to `esnext`. It produces more pure code, and in some cases it matters for tree-shaking.
  The exact case is the following:
  - SFC with 2 `<script lang="ts">` blocks;
  - Some component options:
  
    ```js
    export default {
      inheritAttrs: false
    }
    ```

  As a result, without `esnext` target there is a dirt that prevents component to be tree-shaken.

### CODE REVIEW

File: packages/ui/src/composables/prop-type-filter.ts [hunk 1]
```diff
@@ -13,7 +13,9 @@ export function usePropTypeFilter<T extends object>(props: T) {
       const formattedValue = formatToLog(props[name])
 
       console.warn(
-        `[soramitsu-ui warn]: Invalid prop: type check failed for prop "${name}". Expected: ${formattedList}, got ${formattedValue}`,
+        `[soramitsu-ui warn]: Invalid prop: type check failed for prop "${String(
+          name,
+        )}". Expected: ${formattedList}, got ${formattedValue}`,
       )
```
Comment (0x009922) [inline packages/ui/src/composables/prop-type-filter.ts:19]: `name` var could be a Symbol. TypeScript prevented runtime error!

File: packages/ui/tsconfig.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "paths": {
       "@/*": ["src/*"]
     },
-    
+    "types": ["vitest/importMeta"]
```
Comment (0x009922) [inline packages/ui/tsconfig.json:8]: Make in-source tests possible.

File: packages/ui/vite.config.ts [hunk 3]
```diff
@@ -31,6 +31,12 @@ const vueCompilerTransforms = {
 }
 
 export default defineConfig({
+  test: {
+    include: ['src/**/*.spec.ts'],
+  },
+  define: {
+    'import.meta.vitest': 'undefined',
+  },
```
Comment (0x009922) [inline packages/ui/vite.config.ts:39]: Will be useful when there some in-source tests appear.

**Review by Ctepan — APPROVED** [review]
There are new components, so it's necessary to make same operations to them

**Review by stefashkaa — APPROVED** [review]

**Pull request merged**